orchestrator/dispatcher.go

```diff
@@ -0,0 +1,12 @@
+package orchestrator
+
+import (
+	"io"
+)
+
+type Dispatcher interface {
+	io.Closer
+	Start() <-chan struct{}
+	Handshake(<-chan struct{}) <-chan struct{}
+	Quit() <-chan struct{}
+}
```

Comment:
Just note for future: let's move Dispatcher into its own module in other folder.

Reply:
The interface should be defined where it is used, the implementations of a dispatcher will go in a different folder

Reply:
https://github.com/golang/go/wiki/CodeReviewComments#interfaces